src/librustc_target/abi/mod.rs

```diff
@@ -744,6 +749,12 @@ impl FieldsShape {
     }
 }
 
+/// An identifier that specifies the address space that some operation
+/// should operate on. Special address spaces have an effect on code generation,
+/// depending on the target and the address spaces it implements.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Default)]
+pub struct AddressSpace(pub u32);
```

Comment:
It may be better to avoid `Default` and instead call that method `data`. It'd be shorter at the call sites and potentially more expressive.

Reply:
Assuming that it _is_ the "data" address space...

Reply:
Done in 326b9d8fffd726469577ad0f59b25dadbfe068a9